src/cache/cache_op_executors.rs

```diff
@@ -0,0 +1,105 @@
+use crate::cache::cache::Cache;
+use crate::cache::cache_operations::{CacheResponse, RequestCached};
+use crate::cache::inner_cache::CachedWithCode;
+use crate::utils::errors::{ApiError, ApiResult};
+use rocket::response::content;
+use serde::Serialize;
+use std::time::Duration;
+
+pub const CACHE_REQS_PREFIX: &'static str = "c_reqs";
+pub const CACHE_RESP_PREFIX: &'static str = "c_resp";
+
+pub(super) fn cache_response<S>(
+    cache: &impl Cache,
+    cache_response: &CacheResponse<S>,
+) -> ApiResult<content::Json<String>>
+where
+    S: Serialize,
+{
+    let cache_key = format!("{}_{}", CACHE_RESP_PREFIX, cache_response.key);
+    let cached = cache.fetch(&cache_key);
+    match cached {
+        Some(value) => Ok(content::Json(value)),
+        None => {
+            let resp_string = serde_json::to_string(&cache_response.generate()?)?;
+            cache.create(&cache_key, &resp_string, cache_response.duration);
+            Ok(content::Json(resp_string))
+        }
+    }
+}
+
+pub(super) fn request_cached(
+    cache: &dyn Cache,
+    client: &reqwest::blocking::Client,
+    operation: &RequestCached,
+) -> ApiResult<String> {
+    request_cached_priv(
+        cache,
+        client,
+        &operation.url,
+        operation.cache_duration,
+        operation.error_cache_duration,
+        operation.cache_all_errors,
+        operation.request_timeout,
+    )
+}
+
+fn request_cached_priv(
```

Comment:
Not really a fan of this name, what about renaming the the other method to `request_from_operation` or as this file is call `op_executor` maybe `execute_request_cached`? Could we inline this method? (not sure if this is nicer, so more this is not required ;) )

Reply:
Yeah, I think that makes sense, as it's private anyway adds no value.